src/ray/raylet/local_object_manager.cc

```diff
@@ -402,24 +396,12 @@ void LocalObjectManager::OnObjectSpilled(const std::vector<ObjectID> &object_ids
       continue;
     }
     const auto &worker_addr = freed_it->second.first;
-    auto owner_client = owner_client_pool_.GetOrConnect(worker_addr);
-    RAY_LOG(DEBUG) << "Sending spilled URL " << object_url << " for object " << object_id
-                   << " to owner " << WorkerID::FromBinary(worker_addr.worker_id());
-    owner_client->AddSpilledUrl(
-        request,
-        [object_id, object_url](Status status, const rpc::AddSpilledUrlReply &reply) {
-          // TODO(sang): Currently we assume there's no network failure. We should handle
-          // it properly.
-          if (!status.ok()) {
-            RAY_LOG(DEBUG)
-                << "Failed to send spilled url for object " << object_id
-                << " to object directory, considering the object to have been freed: "
-                << status.ToString();
-          } else {
-            RAY_LOG(DEBUG) << "Object " << object_id << " spilled to " << object_url
-                           << " and object directory has been informed";
-          }
-        });
+    object_directory_->ReportObjectSpilled(object_id,
```

Comment:
This should be thread safe since it's running in the io_service thread. @rkooo567 Could you double check here?

Reply:
Yeah. If this is used within the node manager io service (`main_service_` in object manager), it should be okay. (so I think it should be thread-safe now)